pkg/instrument/dqlite.go

```diff
@@ -0,0 +1,125 @@
+package instrument
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+)
+
+type Listener struct {
+	Network, Address        string
+	AcceptedChan            chan net.Conn
```

Comment:
Do we need the `Chan` suffix? I think it should be clear from usage that this is a channel. Would `AcceptedConns` be better? There's a line later which this change would make read---
```go
l.AcceptedConns <- conn
```
which feels nice